lib/compress.js

```diff
@@ -838,11 +838,12 @@ merge(Compressor.prototype, {
                         if (node instanceof AST_Call
                             || node instanceof AST_Exit
                             || node instanceof AST_PropAccess
-                                && (side_effects || node.has_side_effects(compressor))
+                                && node.expression.may_throw_on_access(compressor)
```

Comment:
This seems insufficient. I think the previous logic was more correct here.

Reply:
What do you think about:
```patch
+                                && (side_effects || node.expression.may_throw_on_access(compressor))
```
or perhaps better still:
```patch
+                                && (side_effects || node.expression.has_side_effects(compressor))
```

Reply:
Yeah I've found some test cases to pinpoint that flaw as well – working on it.

Reply:
Actually I think it ought to be:
```patch
+                                && (side_effects 
+                                    || node.has_side_effects(compressor) 
+                                    || node.expression.may_throw_on_access(compressor))
```

Reply:
`AST_PropAccess.has_side_effects()` contains call to `.expression.may_throw_on_access()`

I incline to just use `may_throw_on_access()` here as the relevant side effects should be picked out by the subsequent `AST_Node.visit()`

Reply:
cool